views/js/admin_order.js

```diff
@@ -54,22 +85,71 @@ var altapay = {
         });
     },
 
+    refund_child_order: function (element) {
+        var namespace = this;
+        var dataArray = [];
+        dataArray.push({name: 'action', value: 'refundRemaining'});
+        dataArray.push({name: 'payment_id', value: $(element).data('payment-id')});
+        dataArray.push({name: 'orderid', value: $(element).data('orderid')});
+        dataArray.push({name: 'remaining_amount', value: $(element).data('remaining_amount')});
+        $.ajax({
+            type: 'POST',
+            url: $(element).data('url'),
+            data: dataArray,
+            success: function (result) {
+                // If result is a JSON string, parse it
+                if (typeof result === 'string') {
+                    try {
+                        result = JSON.parse(result);
+                    } catch (e) {
+                        console.error("Invalid JSON string", e);
+                        jAlert("An error occurred while processing the response.", 'ERROR', function () {
+                            document.location.reload();
+                        });
+                        return;
+                    }
+
+                    // Check the status
+                    namespace.handleResponse(result);
+                }
+            }
+        });
+    },
+
+
     // Perform release
-    release: function (element) {
+    sendemail: function (element) {
+        payment_link = $(element).data('paymentlink');
+
         var namespace = this;
         $.ajax({
             type: 'POST',
+            cache: false,
+            dataType: 'json',
             url: $(element).data('url'),
```

Comment:
How is the authentication being set? Is it the cookies with a session?

Reply:
@emicha , here we are passing the URL which includes a unique security token. This token is specific to the user session which make sure that the Ajax request are legitimate
` $ajaxUrl = $this->context->link->getAdminLink('AdminPayByLink', true) . '&customer_id=' . $orderDetail->id_customer;`